hello_wifi/lib/hello_wifi.ex

```diff
@@ -27,21 +25,52 @@ defmodule HelloWiFi do
     Logger.info("[HelloWiFi] - WiFi Wizard stopped")
   end
 
-  defp maybe_start_wifi_wizard(_wifi_configured? = true) do
-    # By this point we know there is a wlan interface available
-    # and already configured. This would normally mean that you
-    # should then skip starting the WiFi wizard here so that
-    # the device doesn't start the WiFi wizard after every
-    # reboot.
-    #
-    # However, for the example we want to always run the
-    # WiFi wizard on startup. Comment/remove the function below
-    # if you want a more typical experience skipping the wizard
-    # after it has been configured once.
-    VintageNetWizard.run_wizard(on_exit: {__MODULE__, :on_wizard_exit, []})
+  defp maybe_start_wifi_wizard() do
+    with true <- has_wifi?() || :no_wifi,
+         true <- wifi_configured?() || :not_configured,
+         true <- has_networks?() || :no_networks do
+      # By this point we know there is a wlan interface available
+      # and already configured with networks. This would normally
+      # mean that you should then skip starting the WiFi wizard
+      # here so that the device doesn't start the WiFi wizard after
+      # every reboot.
+      #
+      # However, for the example we want to always run the
+      # WiFi wizard on startup. Comment/remove the function below
+      # if you want a more typical experience skipping the wizard
+      # after it has been configured once.
```

Comment:
this comment section is now outdated..

Reply:
I’m not sure I’m following - the behavior hasn’t changed and so this comment is still valid - by this point it is configured with networks, but we start it anyway. Unless I’m missing something?

Reply:
🤦 I was misreading it - garbled up with feedback on the old PR in regards to the button.. but imho it's fine this way - useful if I'm adding the wizard to an existing (already wlan fully config'ed) project.. slightly confusing in a brand new project, or if user is focused on the button thing..